lib/block-supports/elements.php

```diff
@@ -6,14 +6,23 @@
  */
 
 /**
- * Render the elements stylesheet.
+ * Get the elements class names.
+ *
+ * @param  array  $block         Block object.
+ * @return string                The unique class name.
+ */
+function gutenberg_get_elements_class_name ( $block ) {
```

Comment:
I like the idea of programatically generating the class name based on values within the block object, and I think it helps get us a little closer to some of the ideas @youknowriad raised [in the discussion](https://github.com/WordPress/gutenberg/discussions/37495) on improving saving/rendering of block styles (here, it's a good example of sharing a class name between two separate functions).

Since this function will be exposed in the global namespace, is this a good opportunity to create a generic `gutenberg_get_block_support_class_name` function that the other block supports that currently use `uniqid` could also use? We might then have a first parameter of a string prefix, possibly, so each support could add its own `wp-elements-` or `wp-container-` prefix as needed.

With using the entire block object for the MD5, we might only need to use part of the block attributes (e.g. the `style` attribute) rather than the whole thing. That probably doesn't change the implementation here, but it could mean that we could pass this helper function any arbitrary data for generating the unique class name rather than it necessarily being a block object?

I was also wondering about the md5 string, which at 32 characters is pretty long for a class name. Would it cause any issues to use a sub string and just grab the first 16 - 24 characters?

Reply:
I like the idea of having a unique class per block but I think basing it on attributes or content is not necessarily, what we really want is just a unique class for this block any block support can use.

Reply:
Good point, yes I agree, the objective is for a unique id for the instance of the block (in the context of rendering the supports), rather than verifying data integrity of the attributes/content. I wonder how else we might share an id / class name across function calls 🤔